mheiber/TypeScript
# parse private fields - separate syntax kind for PrivateName

## Description of Changes

Parser and Scanner:
- Add new SyntaxKind: SyntaxKind.PrivateName
- Edit baseline reference files to use `¬` (hook) instead of `#` (hash) to test parser error recovery.
- Test for correct parsing of private name declaration and access (`this.#foo`)

Binder and Checker:
- Correctly provide "Private names are not allowed outside class bodies" error
- Test for "Private names are not allowed outside class bodies" error
- "Private names cannot be used as parameters" error and test case

## Steps to Test

- Build the compiler and services (`npm run build:compiler`)
- Tests should pass. Please review the new test files in the `privateNames` directory carefully, the rest are minor tweaks that I hope don't matter.
- Check the AST. Assuming you have a file in the right place with a private name declaration and access, the AST should have private name nodes in both places. 

`node -e "ts = require('./built/local/tsserver');visit = node => {console.log(node); ts.forEachChild(node, visit)};visit(ts.createSourceFile('f', fs.readFileSync('../tssample/src/1.ts', 'utf8')))`


> The binder and checker work for private names in this PR is very primitive, just a start
